_posts/2018-07-30-use-glue-and-athena-with-snowplow-data/index.md

```diff
@@ -0,0 +1,1468 @@
+
+Finally, you can follow either one or both of the [Athena](#use-aws-athena-to-access-the-data) or [Redshift Spectrum](#use-aws-redshift-spectrum-to-access-the-data) steps to achieve that goal.
+
+<h2 id="glue-reqs">2. AWS Glue prerequisites</h2>
+
+Setting up Glue, Glue Data Catalog and their assorted IAM policies is beyond the scope of this guide, but the main prerequisite units are mentioned here along with links to guides to set up everything, before you start following this guide.
+
+As we want to create a schema in some metastore that is shared across all the frameworks we are using, we are going to use the AWS Data Catalog. Also we will need appropriate permissions and aws-cli.
+
+### Setup Data Catalog in Athena
+
+In order to use the created AWS Glue Data Catalog tables in AWS Athena and AWS Redshift Spectrum, you will need to upgrade Athena to use the Data Catalog.
+
+Please familiarize yourself with what that means by reading the [relevant FAQ][athena-dq-faq].
+
+Once you are happy with that change, carry out the upgrade by following the relevant [step-by-step guide][athena-dq-sbs].
+
+### IAM Service Role
+
+You will also need to have an appropriate IAM role for glue that can read from your snowplow archive S3 location and, if following the optional step, write to the output.
+
+Please follow steps 1 and 2 from the AWS Glue getting started guide, which you can find [here][glue-gs1] and [here][glue-gs2].
+
+### Setup AWS Cli
+
+You will also need to have [aws cli][aws-cli] set up, as some actions are going to require it.
+
+Please follow the excellent [AWS documentation][aws-cli] on AWS to get it set-up for your platform, including having the correct credentials with Glue and S3 permissions.
+
+<h2 id="creating-the-source-table-in-aws-glue-data-catalog">3. Creating the source table in AWS Glue Data Catalog</h2>
+
+In order to use the data in Athena and Redshift, you will need to create the table schema in the AWS Glue Data Catalog.
+
+To do that you will need to login to the AWS Console as normal and click on the AWS Glue service. You may need to start typing "glue" for the service to appear:
+
+![Glue click][console-glue-search]
+
+### Creating the database
+
+Then you will need to add a database by clicking on "Databases" from the left pane and then the "Add Database" button:
+
+![Glue add database][glue-add-database]
+
+Set a database name in the dialog (we will use `snowplow_data` as the database name for the rest of this article)
+
+You can also perform that step using the CLI with:
+
+```bash
+aws glue create-database --database-input '{"Name": "snowplow_data", "Description": "Snowplow Data"}'
+```
+
+### Creating the archive table
+
+In order to create the table, you will need to use the CLI as creating all the fields (~130) would be too error-prone and tedious using the console.
+
+Use a command like so:
+
+```bash
```

Comment:
I remember back in August (as you can see from the todo list as well) I suggested that the code listings can go to github gists, but I was told (I don't remember by who), that this will break up the flow. I am happy either way. Are we sure that we want this?